monitoring/procfs.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/gravitational/trace"
+)
+
+// ProcNet* are standard paths to Linux procfs information on sockets
+const (
+	ProcNetTCP  = "/proc/net/tcp"
+	ProcNetTCP6 = "/proc/net/tcp6"
+	ProcNetUDP  = "/proc/net/udp"
+	ProcNetUDP6 = "/proc/net/udp6"
+	ProcNetUnix = "/proc/net/unix"
+)
+
+// SocketState stores Linux socket state
+type SocketState uint8
+
+// Possible Linux socket states
+const (
+	Established SocketState = 0x01
+	SynSent                 = 0x02
+	SynRecv                 = 0x03
+	FinWait1                = 0x04
+	FinWait2                = 0x05
+	TimeWait                = 0x06
+	Close                   = 0x07
+	CloseWait               = 0x08
+	LastAck                 = 0x09
+	Listen                  = 0x0A
+	Closing                 = 0x0B
+)
+
+// TCPSocket contains details on a TCP socket
+type TCPSocket struct {
+	LocalAddress  net.TCPAddr
+	RemoteAddress net.TCPAddr
+	State         SocketState
+	TXQueue       uint
+	RXQueue       uint
+	TimerState    uint
+	TimeToTimeout uint
+	Retransmit    uint
+	UID           uint
+	Inode         uint
+}
+
+// UDPSocket contains details on a UDP socket
+type UDPSocket struct {
+	LocalAddress  net.UDPAddr
+	RemoteAddress net.UDPAddr
+	State         SocketState
+	TXQueue       uint
+	RXQueue       uint
+	UID           uint
+	Inode         uint
+	RefCount      uint
+	Drops         uint
+}
+
+// UnixSocket contains details on a Unix socket
+type UnixSocket struct {
+	RefCount uint
+	Protocol uint
+	Flags    uint
+	Type     uint
+	State    uint
+	Inode    uint
+	Path     string
+}
+
+type socketparser func(string) (interface{}, error)
+
+func parseSocketFile(parse socketparser, fpath string) (interface{}, error) {
```

Comment:
@a-palchikov I've reworked `/proc/net/*` files parsing, not sure if it's a good way since using a lot of `interface{}` and type assertions